source/simulator/assembly.cc

```diff
@@ -646,11 +653,19 @@ namespace aspect
       .connect (std_cxx11::bind(&aspect::Assemblers::MeltEquations<dim>::local_assemble_stokes_preconditioner_melt,
                                 std_cxx11::cref (*melt_equation_assembler),
                                 std_cxx11::_1, std_cxx11::_2, std_cxx11::_3));
-    else
+    else if (!assemble_newton_stokes_system)
```

Comment:
It would feel more natural to me if this would be `if (assemble_newton_stokes_system) ... special stuff ... else ... normal stuff`, is there a reason you used the other way?